client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,169 @@
+    const { filenameDisplay, filetype } = formRef.current.values;
+    const filename = `${filenameDisplay}${filetype}`;
+    dispatch({
+      type: 'DATA_FILES_COMPRESS',
+      payload: { filename, files: selected }
+    });
+  };
+
+  let buttonIcon;
+  if (status === 'RUNNING') {
+    buttonIcon = <LoadingSpinner placement="inline" />;
+  } else if (status === 'ERROR') {
+    buttonIcon = <Icon name="alert" />;
+  } else {
+    buttonIcon = null;
+  }
+  const initialValues = {
+    filenameDisplay:
+      selectedFiles[0] && selectedFiles.length === 1
+        ? selectedFiles[0].name
+        : '',
+    filetype: '.zip'
+  };
+  const validationSchema = yup.object().shape({
+    filenameDisplay: yup
+      .string()
+      .trim('The filename cannot include leading and trailing spaces')
+      .strict(true)
+      .required('The filename is required')
+  });
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpened={onOpened}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <ModalHeader toggle={toggle}>Compress Files</ModalHeader>
+      <Formik
+        innerRef={formRef}
+        initialValues={initialValues}
+        validationSchema={validationSchema}
+        onSubmit={compressCallback}
+      >
+        {({ setFieldValue, values, isValid }) => {
+          const handleSelectChange = e => {
+            setFieldValue('filetype', e.target.value);
+          };
+          const isDisabled = status === 'RUNNING' || status === 'SUCCESS';
+          return (
+            <Form>
+              <ModalBody>
+                <FormField
+                  label="Compressed File Name"
+                  name="filenameDisplay"
+                  disabled={isDisabled}
+                  addonType="append"
+                  addon={
+                    <InputGroupAddon addonType="append">
+                      <Input
+                        type="select"
+                        name="filetype"
+                        bsSize="sm"
+                        onChange={handleSelectChange}
+                        disabled={isDisabled}
+                      >
+                        <option value=".zip">.zip</option>
+                        <option value=".tar.gz">.tar.gz</option>
+                      </Input>
+                    </InputGroupAddon>
+                  }
+                />
+                <p>
+                  A job to compress these files will be submitted. The
+                  compressed file will appear in this directory.
+                </p>
+              </ModalBody>
+              <ModalFooter>
+                <InlineMessage isVisible={status === 'SUCCESS'} type="success">
+                  Successfully started compress job
+                </InlineMessage>
+                <Button
+                  className="data-files-btn"
+                  disabled={isDisabled || !isValid}
+                  styleName="submit-button"
+                  type="submit"
```

Comment:
Hmm, seeing this form able to be submitted on first load for multiple selected files, with no filename yet given. Think we want to disable until there is something there
<img width="469" alt="Screen Shot 2021-01-28 at 4 48 37 PM" src="https://user-images.githubusercontent.com/20326896/106208582-386ccb80-6189-11eb-8bff-08b17efe9308.png">